### ISSUE TITLE
NetworkManager allow disabling default plugins

### ISSUE THREAD

**Issue by jalil-salame** [NixOS/nixpkgs#365713]
## Describe the bug

NetworkManager has a set of default plugins that are enabled:

https://github.com/NixOS/nixpkgs/blob/314e12ba369ccdb9b352a4db26ff419f7c49fa84/nixos/modules/services/networking/networkmanager.nix#L637-L645

These cannot be disabled through the configuration of the module (e.g. to reduce the NixOS image size).

## Steps To Reproduce

N/A

## Expected behavior

The module should instead add the plugins to the `networking.networkmanager.plugins` option. That way they can be turned off if desired:

https://github.com/NixOS/nixpkgs/blob/314e12ba369ccdb9b352a4db26ff419f7c49fa84/nixos/modules/services/networking/networkmanager.nix#L200-L223

## Additional context

NetworkManager's OpenConnect plugin adds about 150MiB to my NixOS image and I don't need it so I was looking to remove it.

## Metadata

```console
❯ nix run nixpkgs#nix-info -- -m
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.63, NixOS, 24.11 (Vicuna), 24.11.20241213.314e12b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.91.1
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux, x86_64-v4-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/jalil/.config/nix/nix.conf:/nix/store/b4q6s25biicxjzxvgqxx5hqc928ira1j-stylix-kde-config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/jalil/.local/share/flatpak/exports/etc/xdg/nix/nix.conf:/var/lib/flatpak/exports/etc/xdg/nix/nix.conf:/home/jalil/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/jalil/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/jalil/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/5fpjpa356qcpljidymbalqxq47yjp9lm-lix-2.91.1/share`
 - nixpkgs: `/nix/store/0xbni69flk8380w0apw4h640n37wn1i9-source
```

## Notify maintainers

@jtojnar 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by jtojnar** [comment]
See https://github.com/NixOS/nixpkgs/issues/137338 and https://github.com/NixOS/nixpkgs/pull/164531

**Comment by jalil-salame** [comment]
Thanks!

Tl;dr: if you want to disable NetworkManager's default plugins, do this:

```nix
{
  networking.networkmanager.plugins = lib.mkForce [ ];
}
```
